fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -790,6 +814,15 @@ public CompletableFuture<RecordCursorContinuation> rebalancePartitions(RecordCur
         }
     }
 
+    private static class RepartitioningContext {
+        boolean emptyingPartition = false;
+        LuceneRecordCursor cursor;
```

Comment:
Would it make sense to have this have the target partition instead, and call `getOldestestNDocuments` or `getNewestNDocuments` based on `removingOldest`, closer to when the cursor is being used.